File: challenges/6-abracadabra/python/guilhermebleidao/abracadabra.py
palavra = input("Digite a palavra: ")
espaços = 0
palavra_tamanho = len(palavra)

for char in palavra:
    print(" " * espaços, end="")
    
    for i in range(palavra_tamanho):
        print(palavra[i], end=" ")
    
    espaços += 1
    palavra_tamanho -= 1
    print()
    

File: challenges/6-abracadabra/js/naygo/abracadabra.js
// Input/Args: palavraBase
// Output: palavra em forma de triângulo

const prompt = require('prompt-sync')()

function abracadabra(palavraBase) {

    let cont = 0

    while(cont <= palavraBase.length) {

        let palavraFormatada = palavraBase.substr(0, (palavraBase.length - cont))
        console.log(' '.repeat(cont) + palavraFormatada.split('').join(' '));

        cont++
    }
}

const verificaEspaco = (string) => string.indexOf(' ') >= 0;

const casosTeste = prompt('Quantos testes deseja fazer: \n=> ')

for(let i = 0; i < casosTeste; i++) {
    
    const palavra = prompt('Digite a mensagem: ')
    
    if(palavra.length > 100 || palavra.length < 2 || verificaEspaco(palavra))
        console.log('A mensagem precisa ter de 2 à 100 caracteres e sem espaços!')
    else
        abracadabra(palavra.toUpperCase())
}